plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -291,8 +291,11 @@ public static Object convert(Object value, Type type)
             // TODO the client sees the bytearray's tostring ouput instead of seeing actual bytes, needs to be fixed.
             return ((ByteBuffer) value).array();
         }
+        if (type instanceof Types.IntegerType || type instanceof Types.DateType) {
+            return Long.parseLong(value.toString());
```

Comment:
There shouldn't be a need to parse the value. Iceberg says both IntegerType and DateType are encoded as `int` so we should just `return value` here. It's better to be strict and correct than having flexibility and incorrectness.

Same for the `TimestampType` branch below. It shouldn't require any change.

Reply:
@hashhar Yes, Iceberg encoded both `IntegerType` and `DateType` as `int`, this function should convert `Iceberg` value to `Trino` value, both `IntegerType` and `DateType` consider `long`, that's why need convert value here.

Reply:
I don't think Long is required. For both `IntegerType` and `DateType` the java class is `Integer`.
See https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/types/Type.java#L31 and https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/types/Type.java#L35.

Your test will still pass if you change this to `return value`.

Reply:
Yes, for `Iceberg` type i think we has agreement on that.
The action convert value to `long` type is, after success read `int` value use `Iceberg` type, we still change it(value) to Trino type, which happens here. 
It can not pass the test when it do not convert to long, because Trino `IntegerType` and `DateType` are encode as `long`. `IntegerType` and `DateType`(Trino) both extends `AbstractIntType`, See https://github.com/trinodb/trino/blob/master/core/trino-spi/src/main/java/io/trino/spi/type/AbstractIntType.java#L45.
After we get converted value, we will final make the value, here will check the value type is compatible with `Trino` or not. See https://github.com/trinodb/trino/blob/master/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsMaker.java#L221 and https://github.com/trinodb/trino/blob/master/core/trino-spi/src/main/java/io/trino/spi/predicate/NullableValue.java#L45